app/forms/search_form.rb

```diff
@@ -66,6 +66,10 @@ def remote_advice_method_ids
     advice_methods.select(&:present?)
   end
 
+  def advice_methods
+    @advice_methods.try(:map) { |e| e.to_i if e.present? }
```

Comment:
There is a whole load of things happening here. Maybe it would be clearer to just `e.to_i` without the conditional, and then `.compact` the result? Also, trying the map feels awkward. Could we not do something like `Array(@advice_methods).map #...` - to ensure it's at the least and empty array and avoid `try`.